src/main/java/com/google/devtools/build/lib/rules/repository/RepositoryDelegatorFunction.java

```diff
@@ -658,8 +658,14 @@ byte[] writeMarkerFile(Map<? extends RepoRecordedInput, String> recordedInputVal
       return new Fingerprint().addString(content).digestAndReset();
     }
 
-    @Nullable
-    byte[] areRepositoryAndMarkerFileConsistent(RepositoryFunction handler, Environment env)
+    sealed interface RepoDirectoryState {
+      record UpToDate(byte[] markerDigest) implements RepoDirectoryState {}
+
```

Comment:
```

[ArrayRecordComponent] Record components should not be arrays.
      record UpToDate(byte[] markerDigest) implements RepoDirectoryState {}
                             ^
    (see http://go/bugpattern/ArrayRecordComponent)
```

Reply:
I ignored this one, but made the interface private to the file.